net/instaweb/rewriter/public/rewrite_options.h

```diff
@@ -1153,6 +1157,26 @@ class RewriteOptions {
     return custom_fetch_headers_.size();
   }
 
+  void AddCustomAsyncUrl(const StringPiece& name);
+
+  const GoogleString* custom_async_url(int i) const {
```

Comment:
Why not just return the vector?  No reason to explode the vector API here.

```c++
const std::vector<GoogleString*>& custom_async_urls() const {
  return custom_async_urls;
}
```

....

Huh, looks like the reasoning is because that's how the surrounding code works.  Ick, ok.